docs/repo/property-pages/property-value-interception.md

```diff
@@ -0,0 +1,160 @@
+﻿# Property Value Interception
+
+Sometimes it is necessary to intercept reads and writes of property values, and modify those values while they are in flight. Property value interception allows this.
+
+Examples of why you may wish to do this:
+
+- You wish to [provide a default value](#providing-default-values) for the property in the UI, when no value is provided in props/targets.
+- The property is a [pseudo property](#pseudo-properties) and should not be written to the project. Its values may be ignored, or may actually be derived from and modify other properties instead.
+
+## Mechanism
+
+The property must opt in to interception. Its `DataSource` must specify the `Persistence` property as one of:
+
+- `ProjectFileWithInterception` writes values to the project file.
+- `UserFileWithInterception` writes values to the project's `.user` file.
+
+For example:
+
+```xml
+<StringProperty Name="MyProperty"
+                DisplayName="My property"
+                ...>
+  <StringProperty.DataSource>
+    <DataSource PersistedName="MyProperty"
+                Persistence="ProjectFileWithInterception"
+                HasConfigurationCondition="False"
+                ... />
+  </StringProperty.DataSource>
+  ...
+</StringProperty>
+```
+
+> See [Property Specification](property-specification.md) for more information on specifying properties.
+
+There must be a corresponding export of `IInterceptingPropertyValueProvider` for the property. For example:
+
+```c#
+[ExportInterceptingPropertyValueProvider("MyProperty", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+internal sealed class MyPropertyValueProvider : InterceptingPropertyValueProviderBase
+{
+    public override Task<string> OnGetUnevaluatedPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
+    {
+        return Task.FromResult("❤️"); // This property's value is always ❤️
+    }
+}
+```
+
+⚠️ The property name passed in the `ExportInterceptingPropertyValueProvider` attribute **must** match the property being intercepted.
+
+The name of the class is not important, but you might like to follow the convention offered here.
+
+A single interceptor class may be used to intercept multiple properties. To do that, add one `ExportInterceptingPropertyValueProvider` attribute per property you wish to intercept, changing the name on each accordingly.
```

Comment:
You can also pass multiple string values in an array to a single `ExportInterceptingPropertyValueProvider` attribute.

Reply:
Thanks, fixed. Actually the attribute has `AllowMultiple = false`, so my previous revision of this paragraph was outright wrong.